pyro/distributions/sine_bivariate_von_mises.py

```diff
@@ -0,0 +1,280 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import warnings
+from copy import copy
+from functools import reduce
+from math import pi
+
+import torch
+from torch.distributions import VonMises
+from torch.distributions.utils import broadcast_all
+
+from pyro.distributions import constraints
+from pyro.distributions.torch_distribution import TorchDistribution
+from pyro.distributions.util import broadcast_shape
+
+
+def _flat_dim(size):
+    return torch.prod(torch.tensor(size), dim=0)
+
+
+def log_I1(orders: int, value: torch.Tensor, terms=250):
+    r""" Compute first n log modified bessel function of first kind
+    .. math ::
+
+        \log(I_v(z)) = v*\log(z/2) + \log(\sum_{k=0}^\inf \exp\left[2*k*\log(z/2) - \sum_kk^k log(kk)
+        - \lgamma(v + k + 1)\right])
+
+    :param orders: orders of the log modified bessel function.
+    :param value: values to compute modified bessel function for
+    :param terms: truncation of summation
+    :return: 0 to orders modified bessel function
+    """
+    orders = orders + 1
+    if len(value.size()) == 0:
+        vshape = torch.Size([1])
+    else:
+        vshape = copy(value.shape)
```

Comment:
Just curious: I believe shapes are immutable, so the need for the `copy(-)` surprises me.  Do you recall the reason?

Reply:
It was to make sure `vshape` wasn't overwritten; however, this is not necessary, as you correctly point out. I'll remove it.